Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,224 @@
+
+        public static string FindRichestPerson(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "rich");
+            var subjects = PeoplePlus(peopleNames);
+            var word = new string[]
+            {
+                (subjects) ? "are" : "is",
+                (subjects) ? "people" : "person"
+            };
+            return $"{peopleNames} {word[0]} the richest {word[1]}. {FormatCash(cash)}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
-        public static string FindMostPoorPerson(string[] peopleAndBalances)
+        public static string FindMostPoorPerson(string[] peoplesBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peoplesBalances))
+
+                return "N/A.";
+
+            var (peopleNames, cash) = CalculatePB(peoplesBalances, "poor");
+            var subjects = PeoplePlus(peopleNames);
+            var word = new string[]
+            {
+                (subjects) ? "have" : "has",
+            };
+
+            return $"{peopleNames} {word[0]} the least money. {FormatCash(cash)}.";
+        }
+
+        private static (StringBuilder peopleNames, float cash) CalculatePB(string[] peoplesBalances, string attribution)
+        {
+            var validation = new string[]
+            { "max", "min", "rich", "poor" };
+
+            if (!validation.Contains(attribution)) return (new StringBuilder(), 0);
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            var peopleNames = new string[peoplesBalances.Length];
+            var cash = new float[peoplesBalances.Length];
+            int iOne, iTwo;
+
+            for (iOne = 0; iOne < peoplesBalances.Length; iOne++)
+            {
+                var peopleInfo = peoplesBalances[iOne].Split(", ");
+                var values = NumPyArrayConversion(peopleInfo[1..]);
+
+                peopleNames[iOne] = peopleInfo[0];
+
+                if (attribution == "max")
+                {
+                    cash[iOne] = values.Max();
+                }
+                else if (attribution == "min")
+                {
+                    if (values.Length <= 1) continue;
+
+                    var balance = new float[values.Length - 1];
+
+                    for (iTwo = 0; iTwo < values.Length - 1; iTwo++)
+                    {
+                        if (iTwo == values.Length - 1) continue;
+                        balance[iTwo] = values[iTwo + 1] - values[iTwo];
+                    }
+                    cash[iOne] = balance.Min();
+                }
+                else if (attribution == "rich" || attribution == "poor")
+                {
+                    cash[iOne] = values[^1];
+                }
```

Comment:
Try changing a design to split function into getting just max or min. Doing both at the same time is not convenient and is much harder to test and reason about. Also add a few comments to make it more clear what is going on.